frontend/src/metabase/components/ModalContent/ModalHeader.module.css

```
@@ -0,0 +1,31 @@
+.HeaderText {
+  font-weight: 700;
+  flex-grow: 1;
+  display: flex;
+  align-items: center;
+  color: var(--mb-color-text-dark);
+
+  &.HeaderTextCentered {
+    justify-content: center;
+  }
+}
+
+.HeaderTextContent {
+  overflow: hidden;
+  text-overflow: ellipsis;
+}
+
+.HeaderTextContainer {
+  flex-grow: 1;
+  /* respect close icon */
+  max-width: calc(100% - 2rem);
```

Close button is rendered conditionally in this component. Luckily we can avoid hardcoding its width:

```suggestion
  overflow: hidden;
```

---

https://github.com/user-attachments/assets/d9ca8c14-e0ff-487c-a6c7-ccb5b41e1643

---

Did you try with `overflow: hidden`?

> but the button itself has padding + negative margin

I believe the negative margin is to balance the padding (to place the icon as if the padding was not there), and the padding is to increase the hit box around the "x" icon - we can ignore it here.

---

yeah, works the same way as with max-width. more or less same solution for me as with max-width, but I'll change it to yours variant 👍

---

0723e14c5e05fa6f0cf211a77705eefb70a64436